src/lib/expression/between_expression.cpp

```diff
@@ -6,27 +6,48 @@
 
 namespace opossum {
 
+static PredicateCondition _get_between_predicate_expression(bool left_inclusive, bool right_inclusive) {
+  if (left_inclusive && right_inclusive) {
+    return PredicateCondition::BetweenInclusive;
+  } else if (left_inclusive && !right_inclusive) {
+    return PredicateCondition::BetweenUpperExclusive;
+  } else if (!left_inclusive && right_inclusive) {
+    return PredicateCondition::BetweenLowerExclusive;
+  } else if (!left_inclusive && !right_inclusive) {
+    return PredicateCondition::BetweenExclusive;
+  }
+  Fail("Unreachable Case");
+}
+
 BetweenExpression::BetweenExpression(const std::shared_ptr<AbstractExpression>& value,
                                      const std::shared_ptr<AbstractExpression>& lower_bound,
-                                     const std::shared_ptr<AbstractExpression>& upper_bound)
-    : AbstractPredicateExpression(PredicateCondition::Between, {value, lower_bound, upper_bound}) {}
+                                     const std::shared_ptr<AbstractExpression>& upper_bound, bool left_inclusive,
```

Comment:
the left exclusive and right exclusive properties need to go from the BetweenExpression. Instread, it should take a PredicateCondition as a constructor parameter and check whether it is a valid between predicate condition